color/src/dynamic.rs

```diff
@@ -447,6 +447,18 @@ impl BitHash for DynamicColor {
     }
 }
 
+// Note that the conversion is only lossless for color spaces that have a tag,
+// this is why we have this additional trait bound. See also
+// https://github.com/linebender/color/pull/155 for more discussion.
+impl<CS: ColorSpace> From<AlphaColor<CS>> for DynamicColor
+where
+    ColorSpaceTag: From<CS>,
+{
+    fn from(value: AlphaColor<CS>) -> Self {
+        Self::from_alpha_color(value)
```

Comment:
Can we have `const { assert!(CS::TAG.is_some()) }` here? This is only a sanity check against mistaken impls of `From`; it's not about making this a potentially fallible impl which would resolve that at monomorphisation time.